### ISSUE TITLE
Contrib suggestion: would you like me to add tests and refactor to TypeScript?

### ISSUE THREAD

**Issue by KemingHe** [dacbd/create-issue-action#37]
Hello @dacbd and @tisonkun , I really like using your action and I want to contribute to making it even better. 

Aside from the README.md touch-up #36 , I also want to explore the option of refactoring the codebase to TypeScript for future-proofing.

In doing so, I can add ACT https://github.com/nektos/act tests and snapshot tests to your work while I refactor. 

Let me know your thoughts in the comments/discussions. Hear from you soon. Happy to help out, always. @KemingHe

**Comment by dacbd** [comment]
@KemingHe I'm more than happy to accept a refactor to typescript. 

I did end up moving tests to separate repo that I essentially test manually.

Mostly I want to have a real e2e test that is in a different repo. (to keep the issue spam from any e2e testing out of this repo I've been doing it here: )

As far as ACT that you linked, I have not personally used it so I would just ask that a PR including it be done separately from a typescript refactor.

**Comment by dacbd** [comment]
here is another example of some more involved actions testing, in this case it was for a comment, but essentially the same can apply for an issue. (a create, verify, delete/clean-up) job/step https://github.com/iterative/cml-playground/blob/main/workflows/cml-1228.yml#L21-L57

**Comment by KemingHe** [comment]
> As far as ACT that you linked, I have not personally used it so I would just ask that a PR including it be done separately from a typescript refactor.

@dacbd Yes, sir! I will respect that instruction and keep you posted on both (TS and ACT) in the coming months.

**Comment by KemingHe** [comment]
<img width="1230" alt="Screenshot 2024-04-25 at 9 19 21 PM" src="https://github.com/dacbd/create-issue-action/assets/44534763/1eabd011-f5ae-487d-a225-9ef2fdb9b72d">

@dacbd Here's a preview of the advanced formatting working in my project. Thank you for creating this awesome and robust tool!

**Comment by dacbd** [comment]
if you want some other inspiration this repo/action is pretty simple and I relatively recently refactored it to typescript: https://github.com/iterative/setup-cml

**Comment by KemingHe** [comment]
Hello @dacbd I'm back to contributing to your project! Had a bit of a break building my new garage. Thanks for the resources. This will help me jump right back in!

**Comment by KemingHe** [comment]
Hi, quick update: life has been busy, progress is slow but I've locally written most of the tests. Will def keep you posted!! @dacbd

**Comment by dacbd** [comment]
very understandable, no worries.

**Comment by KemingHe** [comment]
@dacbd Hi Daniel, I've got a good handle on ACT (https://github.com/nektos/act) and am now able to run/test workflows locally with same behavior as on GitHub. The tool is amazing.

The refactor is still coming; and now it's only going to come sooner with my new found powers. Cheers! Will keep you posted.
